patches/server/0812-Hide-unnecessary-itemmeta-from-clients.patch

```diff
@@ -81,7 +81,13 @@ index 430d4e98134dce62d30ddb31fcb125a69571fa5a..fe7e22d9a69d69dfcce63faa28e90945
 +                if (tag.get("Enchantments") instanceof ListTag enchantmentsTag && !enchantmentsTag.isEmpty()) {
 +                    // The client still renders items with the enchantment glow if the enchantments tag contains at least one (empty) child.
 +                    ListTag enchantments = new ListTag();
-+                    enchantments.add(new CompoundTag());
++                    CompoundTag fakeEnchantment = new CompoundTag();
++                    // Soul speed boots generate client side particles.
++                    if (EnchantmentHelper.getItemEnchantmentLevel(Enchantments.SOUL_SPEED, itemStack) > 0) {
++                        fakeEnchantment.putString("id", "minecraft:soul_speed");
```

Comment:
Just to bring this up before someone merges it, should this reference to the resource loc for soul speed be hardcoded like this? What about nms `Registry.ENCHANTMENT.getKey(Enchantments.SOUL_SPEED)` or bukkit `Enchantment.SOUL_SPEED.getKey()`

Reply:
Thanks. Used the Bukkit variant since it seems to be the fastest (although I doubt that that minuscule difference really matters here)